### PR TITLE
Add the snippets package to the repo.

### PR DESCRIPTION
**Pull request by gspencergoog**
This adds a new package to this repo that allows edit source code samples. It can also create new samples on an existing Dart element.

Eventually, I would like to migrate the snippets tool in the repo to either use the `snippets` package from here, or to just use that package directly via `pub global activate` the way that we do dartdoc.

I have not published the `snippets` package yet, but I will once this PR is committed.

Much of the code in the `snippets` package started out as a copy of the code in the flutter repo's [snippets tool](https://github.com/flutter/flutter/tree/master/dev/snippets) and[ analyze_sample_code.dart](https://github.com/flutter/flutter/blob/master/dev/bots/analyze_sample_code.dart).  This combines them into a single package adds methods for creating new samples and parsing the Dart code with the Dart analyzer's AST parser.

This forms the basis for a follow-on PR that adds a Flutter app that can extract and reinsert samples so that they can be edited, debugged, and run in an IDE.

Sorry for the massive PR...

### CODE REVIEW

**Review by darrenaustin — APPROVED** [review]
LGTM.

Sorry this took so long. I did go through it all, but parts I kind of skimmed. Looks good. Nothing stood out to me other than a couple of comments on the project files.

File: packages/snippets/AUTHORS [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Below is a list of people and organizations that have contributed
+# to the Process project. Names should be added to the list like so:
```
[the remark this answers is not in the document]
Comment (gspencergoog) [reply packages/snippets/AUTHORS:2]: LOL, now you know where I copied it from. :-)

File: packages/snippets/LICENSE [hunk 2]
```diff
@@ -0,0 +1,25 @@
+Copyright 2020 The Flutter Authors. All rights reserved.
```
[the remark this answers is not in the document]
Comment (gspencergoog) [reply packages/snippets/LICENSE:1]: Yes, it matters, it should be the same as the rest of this repo, 2014.

**Comment by gspencergoog** [comment]
@darrenaustin No worries, thanks for the review!

**Pull request merged**